src/funcx_common/messagepack/message_types/ep_status_report.py

```diff
@@ -0,0 +1,77 @@
+import json
+import typing as t
+import uuid
+
+from ..common import Message, MessageType
+from ..exceptions import InvalidMessageError, InvalidMessagePayloadError
+
+
+class EPStatusReport(Message):
+    """
+    Status report for an endpoint, sent from Endpoint to Forwarder.
+
+    Includes EP-wide info such as utilization, as well as per-task status information.
+    """
+
+    message_type = MessageType.EP_STATUS_REPORT
+
+    def __init__(
+        self,
+        endpoint_id: str,
+        ep_status_report: t.Dict[str, t.Any],
+        task_statuses: t.Dict[str, t.Any],
+    ) -> None:
+        if not (isinstance(ep_status_report, dict) and isinstance(task_statuses, dict)):
+            raise InvalidMessageError(
+                "EPStatusReport inner json data was improperly shaped"
+            )
+
+        self.endpoint_id = endpoint_id
+        self.ep_status = ep_status_report
+        self.task_statuses = task_statuses
+
+    @property
+    def endpoint_id_bytes(self) -> bytes:
+        return uuid.UUID(self.endpoint_id).bytes
+
+    def get_v0_body(self) -> bytes:
+        # NOTE: it's important that this is done as a dumps(...).encode("ascii")
+        # rather than the more normal "dumps(..., ensure_ascii=True)"
+        # the reason is that this is the method of encoding used in `funcx-endpoint` and
+        # we must be sure that
+        # 1. messages written by this module can be read there
+        # 2. messages written by that module can be read here
+        #
+        # (1) is ensured by this method of encoding in this function
+        # (2) is ensured by testing that `load_v0_body` is an inverse for this method
+        jsonified = json.dumps(
+            [self.ep_status, self.task_statuses],
+            separators=(",", ":"),
+            sort_keys=True,
```

Comment:
One non-obvious thing in here. `sort_keys=True` makes testing easier because you can do a `json.dumps` call in the tests and then assert equality between the resulting strings.

It might not be strictly necessary for core message passing, but I think it's worth doing for the better testing it lets us do.